### DISCUSSION TITLE
Is a specific Ant version needed to run the tests?

### DISCUSSION THREAD

**Discussion by escay** [eclipse-ee4j/glassfish#25163] [Q&A] [answered]
The build prerequisites mention:
- JDK17+.
- Maven 3.9.0+

But there is no mention of Ant. 
When I want to run

`./runtests.sh jdbc_all`

I see Ant is needed on the command line as well, so I just installed a recent version (1.10.14).
Is it correct that a minimal version of Ant 1.10.12 is assumed to be used?

nucleus\parent\pom.xml defines <ant.version>1.10.12</ant.version>

Should this be added to the build prerequisites? Or should ideally the runtests.sh script use the ant version from maven repo?

Thank you

**Comment by OndroMih** [comment]
As far as I know, Ant is not required for building GlassFish, not even for running the default set of tests that run with `mvn clean install`. The runtests.sh script is only to run the tests locally. In CI, they are executed via Ant artifact downloaded as a maven dependency.

**Reply by escay** [reply]
Clean Ubuntu with only mvn and java 17 installed:
- Building of glassfish succeeds when running `mvn clean install -Pfastest -T4C`
- Running ./runtests.sh script fails

Failure:
```
===== V2 DEVTESTS INIT ===== 

+ find /projects/glassfish/target/glassfish7/glassfish -type f
+ xargs touch
+ echo S1AS_HOME is /projects/glassfish/target/glassfish7/glassfish
S1AS_HOME is /projects/glassfish/target/glassfish7/glassfishI will just
+ echo ANT_HOME is
ANT_HOME is
+ echo APS_HOME is /projects/glassfish/appserver/tests/appserv-tests
APS_HOME is /projects/glassfish/appserver/tests/appserv-tests
+ java -version
openjdk version "17.0.12" 2024-07-16 LTS
OpenJDK Runtime Environment Zulu17.52+17-CA (build 17.0.12+7-LTS)
OpenJDK 64-Bit Server VM Zulu17.52+17-CA (build 17.0.12+7-LTS, mixed mode, sharing)
+ ant -version
./appserver/tests/appserv-tests/devtests/jdbc/../../../common_test.sh: line 54: ant: command not found
```

it's coming from `/projects/glassfish/appserver/tests/common_test.sh`.
For now I will just install ANT.

**Comment by OndroMih** [comment]
Yes, to run ./runtest.sh script you need to install Ant. It’s not executed from mvn install. You don’t need it to build or even run tests which are part of the build. So, installing Ant is not required to build GlassFish. Only to run the runtests script.

**Comment by dmatej** [comment, marked as answer]
The situation is that we are slowly revisiting and transforming these tests (I call them Zombie Test Army) to maven based application-tests. The basic trigger is when some of those tests fails, but there is missing clear output leading to a solution, then usually I learn what that test does and rewrite it so it is executable even in Eclipse IDE, not just by Maven.
However to be able to run these zombies locally, I created the runtests.sh script for now. Basically the latest Ant is a good choice.
